server/player/src/packet_handlers/inventory.rs

```diff
@@ -426,13 +426,66 @@ fn handle_shift_click(
 }
 
 fn handle_number_key(
-    _game: &mut Game,
-    _world: &mut World,
-    _player: Entity,
-    _packet: ClickWindow,
-    _key: u8,
+    game: &mut Game,
+    world: &mut World,
+    player: Entity,
+    packet: ClickWindow,
+    key: u8,
 ) -> anyhow::Result<()> {
-    // TODO
+    let slot: usize = packet.slot as usize;
+    let window = world.get::<Window>(player);
+    let accessor = window.accessor(world)?;
+
+    // The slot index used to index the hotbar
+    let hotbar_slot = SlotIndex {
+        area: Area::Hotbar,
+        slot: (key - 1) as usize,
+    };
+
+    // Convert the slot index into the network slot
```

Comment:
The `accessor` is only used to convert between indices in the protocol and internal indices (`SlotIndex`). Rather than converting `hotbar_slot` to a network slot, you can just call `inventory.item_at` directly with the `Area` and the index within the hotbar.

`WindowAccessor`'s sole purpose is when you have an index from a packet and need to index into the inventory. Otherwise, the APIs directly on `Inventory` should be used.

Reply:
How would I get the `Inventory` in this case? Sorry if I'm missing something obvious.

Reply:
`Inventory` is a component, just like `Window`—just call `World::get::<Inventory>` :)